application-diagram-ui/src/main/resources/Diagram/DiagramSheet.xml

```diff
@@ -634,27 +634,45 @@ define('diagram-graph-xml-filter', ['jquery', 'diagram-config', 'diagram-link-ha
   };
 
   // Replace custom internal links with absolute URLs (but keep some metadata to indicate the entity reference).
+  // Consider both the cases when the node is UserObject or mxCell.
   var onExportLink = function(node) {
-    var link = $(node).attr('link');
+    var tagName = node.tagName.toLowerCase();
+    var value = $(node).attr('value');
+    if (value) {
+      var href = $('&lt;div&gt;&lt;/div&gt;').append(value).find('a[href]').attr('href');
+    }
+    var link = tagName == 'userobject' ? $(node).attr('link') : href;
     if (diagramLinkHandler.isXWikiCustomLink(link)) {
       var url = diagramLinkHandler.getURLFromCustomLink(link);
       var absoluteURL = $('&lt;a/&gt;').attr('href', url).prop('href');
-      $(node).attr({
-        link: absoluteURL,
-        'data-link': link
-      });
+      updateLinkAttributes(node, value, link, absoluteURL);
     }
   };
 
   // Replace absolute URLs with internal custom links, if the target entity reference is indicated.
   var onImportLink = function(node) {
     var link = $(node).attr('data-link');
     if (diagramLinkHandler.isXWikiCustomLink(link)) {
-      $(node).attr({
-        link: link,
+      var value = $(node).attr('value');
+      if (value) {
+        var href = $('&lt;div&gt;&lt;/div&gt;').append(value).find('a[href]').attr('href');
+      }
+      updateLinkAttributes(node, value, href, link);
+    }
+  };
+
+  var updateLinkAttributes = function(node, value, oldURL, newURL) {
+    var attributesMap = {
+      'link': newURL,
+      'data-link': null
+    };
+    if (node.tagName.toLowerCase() != 'userobject') {
+      attributesMap = {
+        'value': value.replace(oldURL, newURL),
         'data-link': null
```

Comment:
I would initialize ``attributesMap`` only with ``data-link`` (since it's shared) and then set the extra property on if / else.

Reply:
Indeed, it looks cleaner